test/gurka/test_matrix.cc

```diff
@@ -0,0 +1,46 @@
+#include "gurka.h"
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+TEST(MatrixTest, MatrixSimple) {
+  constexpr double gridsize = 100;
+
+  // the grid below is 50 x 100 meters at 36 km/h, i.e. 10 m/s
+  const std::string ascii_map = R"(
+      A----B
+      |    |
+      |    |
+      C----D
+    )";
```

Comment:
Just a small dumb test to show it works. Unfortunately, there's nothing relevant inside the `Api` object that gets returned from `do_action` as we don't track stuff in PBF for matrix. We can only predict the `distance` but that could be fair enough with the right map